File: php/features.php
<?php
declare(strict_types=1);

// Vdraw SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class VdrawFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new VdrawBaseFeature();
            case "test":
                return new VdrawTestFeature();
            default:
                return new VdrawBaseFeature();
        }
    }
}

File: py/features.py
# Vdraw SDK feature factory

from feature.base_feature import VdrawBaseFeature
from feature.test_feature import VdrawTestFeature


def _make_feature(name):
    features = {
        "base": lambda: VdrawBaseFeature(),
        "test": lambda: VdrawTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
